api/image.go

```diff
@@ -443,21 +440,17 @@ func (api *API) UpdateDownloadHandler(w http.ResponseWriter, req *http.Request)
 		return
 	}
 
-	// create image update with the allowed field. Public, state, href, error or timepstamps can't be provided by the caller.
-	now := time.Now()
-	imageUpdate := &models.Image{
-		Downloads: map[string]models.Download{
-			variant: {
-				Size:            download.Size,
-				Palette:         download.Palette,
-				Type:            download.Type,
-				Width:           download.Width,
-				Height:          download.Height,
-				Private:         download.Private,
-				State:           models.StateDownloadImported.String(),
-				ImportCompleted: &now,
-			},
-		},
+	// Validate a possible mismatch of variant id
+	if download.ID != "" && download.ID != variant {
+		handleError(ctx, w, apierrors.ErrVariantIDMismatch, logdata)
+		return
+	}
+	download.ID = variant
+
+	// generic download validation
+	if err := download.Validate(); err != nil {
+		handleError(ctx, w, err, logdata)
+		return
```

Comment:
I wasn't able to see any tests covering this validation, not sure if this was intentional or not.

Reply:
Good spot, I copied this from the update image endpoint last minute and adding this test showed I hadn't translated the error code into a http status 400 either. Added test and translation accordingly.